doximity/es-elasticity
# Prep work for Elasticsearch 8 upgrade

Jira issue link: [MOFONETNAV-167]

Overview
--------
Updating test suite and removing outdated specs in preparation for Elasticsearch 8 support.
- Use `refresh` instead of `flush` where search is expected to be updated
- Drop support for Ruby 2.5 and 2.6
- Add support for Ruby 2.7 and 3.0, build against Ruby 3.1
- Set `search_type` to `query_then_fetch` as we no longer support Elasticsearch version < 7.0
- Set `include_type_name` when fetching the mapping. Will be removed in the next version to support ES version 8.
- Index names automatically use the new timestamp format when creating an index.

QA Instructions
---------------
See https://github.com/doximity/doximity/pull/39531

[MOFONETNAV-167]: https://doximity.atlassian.net/browse/MOFONETNAV-167?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
